open_spiel/games/2048.h

```diff
@@ -0,0 +1,144 @@
+    * kChanceTiles.size();
+
+struct Coordinate {
+  int x, y;
+  Coordinate(int _x, int _y)
+      : x(_x), y(_y) {}
+};
+
+struct ChanceAction {
+  int row;
+  int column;
+  bool is_four;
+  ChanceAction(int _row, int _column, bool _is_four)
+      : row(_row),
+        column(_column),
+        is_four(_is_four) {}
+};
+
+struct Tile {
+  int value;
+  bool is_merged;
+  Tile(int _value, bool _is_merged)
+      : value(_value),
+        is_merged(_is_merged) {}
+};
+
+// State of an in-play game.
+class TwoZeroFourEightState : public State {
+ public:
+  explicit TwoZeroFourEightState(std::shared_ptr<const Game> game);
+  Player CurrentPlayer() const override {
+    return IsTerminal() ? kTerminalPlayerId : current_player_;
+  }
+  std::string ActionToString(Player player, Action action_id) const override;
+  std::string ToString() const override;
+  bool IsTerminal() const override;
+  std::vector<double> Returns() const override;
+  std::string InformationStateString(Player player) const override;
+  std::string ObservationString(Player player) const override;
+  void ObservationTensor(Player player,
+                         absl::Span<float> values) const override;
+  std::unique_ptr<State> Clone() const override {
+    return std::unique_ptr<State>(new TwoZeroFourEightState(*this));
+  }
+  void UndoAction(Player player, Action action) override;
+  bool InBounds(int row, int column) const;
+  void SetCustomBoard(const std::vector<int> board_seq);
+  ChanceAction SpielActionToChanceAction(Action action) const;
+  Action ChanceActionToSpielAction(ChanceAction move) const;
+  void SetBoard(int row, int column, Tile tile) {
+    board_[row * kDefaultColumns + column] = tile;
+  }
+  Tile BoardAt(int row, int column) const {
+    return board_[row * kDefaultColumns + column];
+  }
+  std::vector<Action> LegalActions() const override;
+  ActionsAndProbs ChanceOutcomes() const override;  
+  int AvailableCellCount() const;
+  std::vector<std::vector<int>> BuildTraversals(int direction) const;
+  bool WithinBounds(int x, int y) const;
+  bool CellAvailable(int x, int y) const;
+  std::vector<Coordinate> 
+      FindFarthestPosition(int x, int y, int direction) const;
+  bool TileMatchesAvailable() const;
+  bool Reached2048() const;
+  void PrepareTiles();
+  int GetCellContent(int x, int y) const;
+
+ protected:
+  void DoApplyAction(Action action) override;
+
+ private:
+  Player current_player_ = kChancePlayerId;
+  std::vector<Tile> board_;
+  bool extra_chance_turn_ = true;
+};
+
+// Game object.
+class TwoZeroFourEightGame : public Game {
+ public:
+  explicit TwoZeroFourEightGame(const GameParameters& params);
+  int NumDistinctActions() const override;
+  std::unique_ptr<State> NewInitialState() const override {
+    return absl::make_unique<TwoZeroFourEightState>(shared_from_this());
+  }
+  int NumPlayers() const override { return kNumPlayers; }
+  double MinUtility() const override { return -1; }
```

Comment:
These don't seem correct for this game?

(min utility should be 0, right? Max utility will depend on a few things)

Reply:
Have changed this